Modeller.fs

```diff
@@ -454,27 +453,28 @@ let coreSemantics =
 let rec modelExpr
   (env : VarMap)
   (varF : Var -> 'var)
-  : Expression -> Result<Expr<Sym<'var>>, ExprError> =
-    function
+  (e : Expression) : Result<Expr<Sym<'var>>, ExprError> =
+    match e.Node with
```

Comment:
Yeah, I went a bit pointfree at this point, it might be worth making things explicit again since the pointfree-ness actually changes the semantics of these functions in strange and unexpected ways